_posts/2021-09-06-terraform-uncommon-options.markdown

```diff
@@ -0,0 +1,336 @@
+---
+excerpt: "Once a new technology rolls over you, if you're not part of the steamroller, you're part of the road."
+layout: page
+title: Terraform, The Less-Common Commands And Options
+permalink: terraform-the-less-common-commands-and-options
+tags: terraform commands console uncommon console cli cloud
+published: true
+---
+
+<p align="center">
+  <a href="https://unsplash.com/photos/Yraw_pJcAoo">
+    <img alt="desert" title="Desert" src="assets/images/2021_09_planet_earth.jpg" align="middle" width="600" height="350" style="border:1px solid #2C4D56">
+  </a>
+</p>
+<br/>
+
+> "Once a new technology rolls over you, if you're not part of the steamroller, you're part of the road." - Stewart Brand
+
+<br/>
+
+The most frequently used Terraform commands, **plan**, **apply**, and **destroy** are the first things I learned when I started using Terraform. Primarily because those commands are a crucial part of Terraform’s cycle. Over time, as Terraform and its community grew, the features of Terraform grew in parallel. Now there are a lot more commands, options, arguments, and functions. Some are not used as often, and some do not even show when invoking the help menu.
+
+In this blog post, I want to talk about the not-so-very common commands, arguments, and use cases.
+
+
+## [Terraform Console]
+
+This command is my current personal fav, the **console** command. As per documentation it is most commonly used for experimenting with expressions and functions. As an added extra, if the command is invoked in a folder with an active Terraform state it will read it from the configured backend. Inspecting and experimenting with the state with Terraform's console can be fun!
+
+Getting to it is as simple as typing in the command line:
+
+```sh
+$ terraform console
+```
+
+Now we can start "playing" with our state, provided there is something in it that we can fiddle with.
+
+For the sake of this blog post let's create a `main.tf` file with the following contents. The file is also available in my [examples repo].
+
+```hcl
+locals {
+  map = {
+    a = 1
+    b = 2
+    c = 3
+  }
+}
+
+resource "local_file" "private_key" {
+  content  = jsonencode(local.map)
+  filename = "./map.json"
+}
+```
+Needless to say, executing `terraform init` is necessary to pull the plugins and initialize the configured backend if there is any.
+
+Afterwads running `terraform apply` against this template, a JSON encoded file with the content of the local variable `map` will be created in the same folder. That is easy enough.
+
+But what if I needed a YAML encoded file? I am also not sure how it will look alike. I prefer JSON, but I also like to have some comments.
```

Comment:
I'd remove "alike", it's not needed when you have "how it will look" before it. The phrase you're looking for in this sentence is "I am also not sure how it will look."